### DISCUSSION TITLE
No FileUpload, No FileRemover, No MQTTconfig

### DISCUSSION THREAD

**Discussion by dozwin81** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#511] [Q&A]
Hi, i have a Helsinki with a buying Chip and „Firmware version: 2022-10-08 Model: MIAMI2021 „
 , after i config the MQTT to my iObroker the Helsinki sent infos but he also reboot after few minutes and i cant on the webUI, this return every 5minuts.

I have deaktivate the mqtt broker at iobroker and manage ist manuell!

at this time i have read from V4.1.2 and i want to update, but i cant go to FileUplaod, FileRemover and MQTT config!

How can i Update?

![IMG_6296](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/c900c386-e70c-4de7-8f23-f722a163fc52)
![IMG_6295](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/0d0be057-62e8-44f3-85b7-6044057dbaf4)
![IMG_6294](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/227030b7-3cfc-419e-a5d1-bc212f2f28b2)
![IMG_6293](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/a0c67212-1104-4282-bdac-b896c71e4500)

**Comment by visualapproach** [comment]
I'm not sure what I'm looking at here. Broken webpages? What's your previous version? Have you tried to update to v4.1.2 and it failed, or do you ask how to update?

**Reply by dozwin81** [reply]
My Question, how to update?

The Pics are at this time for

1Pic: MQTT config crashed
2Pic: Direktory
3Pic: File Upload
4Pic: File Remover

**Reply by dozwin81** [reply]
My Verion is: 2022-10-08

**Reply by visualapproach** [reply]
Okay, you download the project zip file. Uncompress and open code folder in platformio. If connected with usb just press the arrow at the bottom of the screen: "build and upload". Then go to platformio icon and press build file system image. Then upload file system image. Done. Now configure esp via web ui

**Comment by dozwin81** [comment]
It dosent work, I open the code folder, I connect via USB and click Build and than I click upload, but upload has an error, see the pic!

![Bildschirmfoto 2023-05-17 um 18 03 36](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/962d9bdc-3149-476c-a322-10b0d6028870)

**Reply by visualapproach** [reply]
Ok so it can't connect to the com port. I'm no Mac expert but:
Make sure you have a DATA usb cable. There is a lot of cheapo charger cables that only has two wires of four inside.
Some ESP:s has to be forced into flash mode by holding the flash button, press the reset button and  release reset then flash. This should be done just prior to uploading.
I'm on windows 11 and have had some recent issues with the drivers for devices with CH340 chip on it (instead of CP2101 or whatever it's called)
If you have a working firmware on the ESP you could upload OTA instead by editing platformio.ini

**Reply by dozwin81** [reply]
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/ddd43d7e-d20b-409b-a739-95169202e3e7)

**Reply by dozwin81** [reply]
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/7e71bac9-e83e-4f5c-8955-e0d58270bedf)

**Comment by dozwin81** [comment]
OK, after i install the CH340 driver I have connection to the pcb. But after Build and Upload I can see the new Firmware at the home of the web ui, after new connecting in the bestway it doesn't start with sound, no works, no new wifi ap no everything.
![Bildschirmfoto 2023-05-17 um 20 09 59](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/401dbab6-530e-46d5-933e-e11da1b1e8ca)
![Bildschirmfoto 2023-05-17 um 20 10 08](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/cff6038f-aa49-49d5-99d1-bd8308b4faec)

**Comment by dozwin81** [comment]
OK I take it!

After it dosent work I have:
Erase Flash
Build File
Upload File
Upload General

and I have new wifi

**Reply by dozwin81** [reply]
Sorry for the next quetion 🙈

Which ist the Right Hardware Config, it doesent peep on start and the display is blinking whith the Wifi led 🤷‍♂️😩

**Reply by dozwin81** [reply]
https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/119894910/371eca5b-dd3e-40f5-8d0b-8f53e7ec3fc0

**Comment by visualapproach** [comment]
There is only three or four 6-wire models to choose from. Try them out and see which is working.
